aws-configure-credentials-oidc/orb.yml

```diff
@@ -0,0 +1,22 @@
+version: 2.1
+description: An orb to authenticate with AWS using CircleCI OIDC
+orbs:
+  aws-cli: circleci/aws-cli@2.1.0
+commands:
+  aws-configure-credentials:
+    description: "Assume the specified role using the OIDC token"
+    parameters:
+      role-arn:
+        description: The ARN of the IAM Role to assume (e.g. arn:aws:iam::999999999999:role/CircleCI).
+        type: string
+      profile-name:
+        default: default
+        description: Profile name to be configured.
+        type: string
+    steps:
+      - aws-cli/install
+      - run: |
+          STS=($(aws sts assume-role-with-web-identity --role-arn << parameters.role-arn >> --role-session-name "CircleCI-${CIRCLE_PROJECT_REPONAME}-${CIRCLE_BUILD_NUM}" --web-identity-token $CIRCLE_OIDC_TOKEN --duration-seconds 3600 --query 'Credentials.[AccessKeyId,SecretAccessKey,SessionToken]' --output text))
```

Comment:
Might be worth making the duration of the role assumption configurable? If people are using this to apply Terraform, some resource creation can take a while. (3600 seconds seems a good default though)

Reply:
Good point. One hour should be sufficient. It think for initial deployments (where there are changes required to route53-as-code it could fail and need to be re-run). Will follow up with a PR to make this configurable.